packages/runtime/runtime-utils/src/summarizerNode/summarizerNode.ts

```diff
@@ -159,12 +192,21 @@ export class SummarizerNode implements IRootSummarizerNode {
 			}
 		}
 
-		const result = await this.summarizeInternalFn(fullTree, true, telemetryContext);
-		this.wipLocalPaths = { localPath: EscapedPath.create(result.id) };
-		if (result.pathPartsForChildren !== undefined) {
-			this.wipLocalPaths.additionalPath = EscapedPath.createAndConcat(
-				result.pathPartsForChildren,
-			);
+		const result = await scheduleIdleTask(async (deadline: IdleDeadline) => {
+			if (deadline.didTimeout) {
+				// report how often it happens, to possibly adjust abount of itme we allow to wait for idle
+				this.logger.sendErrorEvent({ eventName: "SummarizeHitIdleTimeout" });
+			}
+			return this.summarizeInternalFn(fullTree, trackState, telemetryContext);
+		}, 5000); // 5 seconds max timeout - if CPU is not idle that long, we screwed up responsivness anyway, time to continue with summary
```

Comment:
The part that I do not like about this - this potential delay will happen at every depth of the summary tree.
In current system it feels like it needs to apply only to leaf nodes (DDSs) - where actual work happens.
But in terms of future-proofing, this seems like a wrong direction, as maybe we will have arbitrary length of the trees (with nested channels) and complexity might not be concentrated in leaf nodes (though if feels even if it's the case - most likely yielding only at leaf nodes would be Ok even then). Not sure - happy to hear feedback.